### DISCUSSION TITLE
Minimum system requirements for Ankaios system for arm64 platform

### DISCUSSION THREAD

**Discussion by pinakin2050** [eclipse-ankaios/ankaios#188] [Q&A] [answered]
Hello,
I want to know the minimum system requirements for Ankaios system for arm64 platform. Also, if anyone could tell me the system usage like CPU usage, Memory Usage & number of threads it spawns it would be really helpful.

**Comment by windsource** [comment, marked as answer]
Hi @pinakin2050,

when using Ankaios on arm64 platforms like RaspberryPI 4 with 4GB of RAM it works fine. But probably also smaller footprints with (much) less memory should work.

I have just checked the resource consumption of v0.2.0.

If there is no workload running we get:

```shell
❯ ps -o pid,%cpu,%mem,rss,nlwp,cmd -C ank-server,ank-agent
    PID %CPU %MEM   RSS NLWP CMD
   7258  0.0  0.0  6104    9 /usr/local/bin/ank-server --startup-config /etc/ankaios/state.yaml
   7307  0.2  0.0  6376    9 /usr/local/bin/ank-agent --name agent_A
```

So Ankaios server uses 6.0MB of memory with 9 threads and the agent 6.2MB with 9 threads. Both use 0.0% of CPU time.

When we start nginx as workload we get:

```shell
❯ ank get workloads
 WORKLOAD NAME   AGENT     RUNTIME   EXECUTION STATE
 nginx           agent_A   podman    Running

❯ ps -o pid,%cpu,%mem,rss,nlwp,cmd -C ank-server,ank-agent
    PID %CPU %MEM   RSS NLWP CMD
   7258  0.0  0.0  6232    9 /usr/local/bin/ank-server --startup-config /etc/ankaios/state.yaml
   7307  0.0  0.0  6632   11 /usr/local/bin/ank-agent --name agent_A
```

So now Ankaios server uses 6.1MB of memory with 9 threads and the agent 6.5MB with 11 threads. Still both use 0.0% of CPU time.

**Reply by pinakin2050** [reply]
Thanks for such a detailed answer.
